language_detector/main.py

```diff
@@ -1,9 +1,35 @@
 """This is the entry point of the program."""
+# coding=utf-8
 
 from .languages import LANGUAGES
 
-
 def detect_language(text, languages=LANGUAGES):
     """Returns the detected language of given text."""
-    # implement your solution here
-    pass
+    text = text.strip().split("\n\n")
+    count = {}
+    
+    for t in text:
+        count[find_lang(t, languages=LANGUAGES)] = len(t)
+        
+    return max(count, key=count.get)
```

Comment:
So if I'm understanding correctly, the solution you've implemented breaks and cleans up the text, checks if a paragraph is a specific language by finding if the most frequent words are in a specific languages' common words and then assigns the full length of that paragraph to the count dictionary under that found language. The program then returns the language with the highest number.

The only issue I can see with this kind of solution is the assigning of the full length of the paragraph towards a specific language.For instance, if I have 51 German words and 50 English words, all 101 words get assigned to German (aka it's considered a German paragraph).

A potential option might be to simply return how many words of each language are found in `find_lang` and add each of those to the count dict for a more accurate result.